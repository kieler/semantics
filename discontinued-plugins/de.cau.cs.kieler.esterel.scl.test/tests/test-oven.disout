module Oven:
input START;
input STOP;
input DOOROPEN;
input MINUTE;
output HEAT;
output LIGHT;
output BEEP;

signal MINUTE2 in
    loop
      trap T in
        [
          abort
            loop
              pause
            end loop
          when immediate START;
          abort
            abort
              [
                loop
                  emit HEAT;
                  pause
                end loop
              ||
                loop
                  emit LIGHT;
                  pause
                end loop
              ]
            when immediate [DOOROPEN or STOP]
          when MINUTE2 do
            emit BEEP
          end abort;
          exit T
        ||
          loop
            abort
              loop
                pause
              end loop
            when MINUTE;
            abort
              loop
                pause
              end loop
            when MINUTE;
            emit MINUTE2
          end loop
        ]
      end trap;
      abort
        loop
          pause
        end loop
      when tick
    end loop
  ||
    every immediate DOOROPEN do
      abort
        loop
          emit LIGHT;
          pause
        end loop
      when [not DOOROPEN]
    end every
end signal

end module
