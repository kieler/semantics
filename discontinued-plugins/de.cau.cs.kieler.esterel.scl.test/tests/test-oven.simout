   0 HEAT=0 LIGHT=1 BEEP=0 
   1 HEAT=0 LIGHT=1 BEEP=0 
   2 HEAT=0 LIGHT=0 BEEP=0 
   3 HEAT=0 LIGHT=1 BEEP=0 
   4 HEAT=0 LIGHT=0 BEEP=0 
   5 HEAT=0 LIGHT=1 BEEP=0 
   6 HEAT=0 LIGHT=0 BEEP=0 
   7 HEAT=0 LIGHT=0 BEEP=0 
   8 HEAT=0 LIGHT=1 BEEP=0 
   9 HEAT=0 LIGHT=1 BEEP=0 
  10 HEAT=0 LIGHT=1 BEEP=0 
  11 HEAT=0 LIGHT=1 BEEP=0 
  12 HEAT=0 LIGHT=1 BEEP=0 
  13 HEAT=0 LIGHT=0 BEEP=0 
  14 HEAT=1 LIGHT=1 BEEP=0 
  15 HEAT=1 LIGHT=1 BEEP=0 
  16 HEAT=0 LIGHT=1 BEEP=0 
  17 HEAT=0 LIGHT=0 BEEP=0 
  18 HEAT=0 LIGHT=1 BEEP=0 
  19 HEAT=0 LIGHT=0 BEEP=0 
  20 HEAT=0 LIGHT=1 BEEP=0 
  21 HEAT=0 LIGHT=1 BEEP=0 
  22 HEAT=0 LIGHT=0 BEEP=0 
  23 HEAT=0 LIGHT=1 BEEP=0 
  24 HEAT=0 LIGHT=0 BEEP=0 
  25 HEAT=0 LIGHT=0 BEEP=0 
  26 HEAT=0 LIGHT=1 BEEP=0 
  27 HEAT=0 LIGHT=1 BEEP=0 
  28 HEAT=0 LIGHT=0 BEEP=0 
  29 HEAT=1 LIGHT=1 BEEP=0 
  30 HEAT=0 LIGHT=1 BEEP=0 
  31 HEAT=0 LIGHT=1 BEEP=0 
  32 HEAT=0 LIGHT=1 BEEP=0 
  33 HEAT=0 LIGHT=1 BEEP=0 
  34 HEAT=0 LIGHT=0 BEEP=0 
  35 HEAT=0 LIGHT=0 BEEP=0 
  36 HEAT=0 LIGHT=1 BEEP=0 
  37 HEAT=1 LIGHT=1 BEEP=0 
  38 HEAT=0 LIGHT=0 BEEP=0 
  39 HEAT=0 LIGHT=1 BEEP=0 
  40 HEAT=0 LIGHT=1 BEEP=0 
  41 HEAT=0 LIGHT=0 BEEP=0 
  42 HEAT=0 LIGHT=1 BEEP=0 
  43 HEAT=0 LIGHT=1 BEEP=0 
  44 HEAT=0 LIGHT=1 BEEP=0 
  45 HEAT=0 LIGHT=1 BEEP=0 
  46 HEAT=0 LIGHT=1 BEEP=0 
  47 HEAT=1 LIGHT=1 BEEP=0 
  48 HEAT=0 LIGHT=1 BEEP=0 
  49 HEAT=0 LIGHT=1 BEEP=0 
  50 HEAT=0 LIGHT=0 BEEP=0 
  51 HEAT=0 LIGHT=0 BEEP=0 
  52 HEAT=0 LIGHT=0 BEEP=0 
  53 HEAT=0 LIGHT=1 BEEP=0 
  54 HEAT=0 LIGHT=1 BEEP=0 
  55 HEAT=0 LIGHT=0 BEEP=0 
  56 HEAT=0 LIGHT=1 BEEP=0 
  57 HEAT=0 LIGHT=1 BEEP=0 
  58 HEAT=0 LIGHT=1 BEEP=0 
  59 HEAT=0 LIGHT=0 BEEP=0 
  60 HEAT=0 LIGHT=0 BEEP=0 
  61 HEAT=0 LIGHT=0 BEEP=0 
  62 HEAT=0 LIGHT=1 BEEP=0 
  63 HEAT=1 LIGHT=1 BEEP=0 
  64 HEAT=1 LIGHT=1 BEEP=0 
  65 HEAT=1 LIGHT=1 BEEP=0 
  66 HEAT=0 LIGHT=0 BEEP=0 
  67 HEAT=1 LIGHT=1 BEEP=0 
  68 HEAT=1 LIGHT=1 BEEP=0 
  69 HEAT=0 LIGHT=0 BEEP=0 
  70 HEAT=0 LIGHT=1 BEEP=0 
  71 HEAT=0 LIGHT=0 BEEP=0 
  72 HEAT=0 LIGHT=0 BEEP=0 
  73 HEAT=0 LIGHT=0 BEEP=0 
  74 HEAT=1 LIGHT=1 BEEP=0 
  75 HEAT=1 LIGHT=1 BEEP=0 
  76 HEAT=0 LIGHT=1 BEEP=0 
  77 HEAT=0 LIGHT=0 BEEP=0 
  78 HEAT=0 LIGHT=1 BEEP=0 
  79 HEAT=0 LIGHT=1 BEEP=0 
  80 HEAT=0 LIGHT=1 BEEP=0 
  81 HEAT=0 LIGHT=1 BEEP=0 
  82 HEAT=0 LIGHT=1 BEEP=0 
  83 HEAT=0 LIGHT=0 BEEP=0 
  84 HEAT=0 LIGHT=1 BEEP=0 
  85 HEAT=0 LIGHT=1 BEEP=0 
  86 HEAT=0 LIGHT=1 BEEP=0 
  87 HEAT=0 LIGHT=0 BEEP=0 
  88 HEAT=0 LIGHT=1 BEEP=0 
  89 HEAT=0 LIGHT=0 BEEP=0 
  90 HEAT=0 LIGHT=0 BEEP=0 
  91 HEAT=0 LIGHT=1 BEEP=0 
  92 HEAT=0 LIGHT=1 BEEP=0 
  93 HEAT=0 LIGHT=0 BEEP=0 
  94 HEAT=1 LIGHT=1 BEEP=0 
  95 HEAT=0 LIGHT=1 BEEP=0 
  96 HEAT=0 LIGHT=1 BEEP=0 
  97 HEAT=0 LIGHT=1 BEEP=0 
  98 HEAT=0 LIGHT=0 BEEP=0 
  99 HEAT=0 LIGHT=0 BEEP=0 
