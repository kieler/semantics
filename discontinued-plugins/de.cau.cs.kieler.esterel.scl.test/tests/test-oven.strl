module Oven:
        input START, STOP, DOOROPEN,  MINUTE;
        output HEAT, LIGHT, BEEP;

        signal MINUTE2 in
        loop
            trap T in
               [await immediate START; 
                 [ abort
                        abort      
                                   sustain HEAT
				|| sustain LIGHT
				
                       when immediate [(DOOROPEN or STOP)]; 
                 when  MINUTE2 do
                          emit BEEP
		 end abort; exit T      
                 ]
	       ||
		 loop 
		     await MINUTE; 
		     await MINUTE;     
		     emit  MINUTE2   
		 end loop
		]
            end;
	    await tick
        end loop

        ||
        every  immediate DOOROPEN do     
                abort
                        sustain LIGHT
                when [not DOOROPEN]
        end every
        end signal
end module
