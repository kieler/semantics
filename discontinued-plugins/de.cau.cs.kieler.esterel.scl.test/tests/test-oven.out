module Oven:
input START;
input STOP;
input DOOROPEN;
input MINUTE;
output HEAT;
output LIGHT;
output BEEP;

signal MINUTE2 in
    loop
      trap T in
        [
          await immediate START;
          abort
            abort
              [
                sustain HEAT
              ||
                sustain LIGHT
              ]
            when immediate [DOOROPEN or STOP]
          when MINUTE2 do
            emit BEEP
          end abort;
          exit T
        ||
          loop
            await MINUTE;
            await MINUTE;
            emit MINUTE2
          end loop
        ]
      end trap;
      await tick
    end loop
  ||
    every immediate DOOROPEN do
      abort
        sustain LIGHT
      when [not DOOROPEN]
    end every
end signal

end module
